src/lib/y2partitioner/dialogs/format_and_mount.rb

```diff
@@ -7,28 +7,42 @@ module Dialogs
     # Part of {Sequences::AddPartition} and {Sequences::EditBlkDevice}.
     # Formerly MiniWorkflowStepFormatMount
     class FormatAndMount < CWM::Dialog
-      # @param partition [Y2Storage::Partition] FIXME: unsure which type we want
-      def initialize(partition)
-        @partition = partition
+      # @param options [Y2Partitioner::FormatMountOptions]
+      def initialize(options)
         textdomain "storage"
 
-        @format_widget = Widgets::FormatOptions.new(@partition)
-        @mount_widget  = Widgets::MountOptions.new(@partition)
+        @options = options
       end
 
       def title
-        "Edit Partition #{@partition.name}"
+        "Edit Partition #{@options.name}"
       end
 
       def contents
         HVSquash(
           HBox(
-            @format_widget,
-            HSpacing(4),
-            @mount_widget
+            Widgets::FormatOptions.new(@options),
+            HSpacing(5),
+            Widgets::MountOptions.new(@options)
           )
         )
       end
+
+      def cwm_show
+        ret = nil
+
+        loop do
+          ret = super
+
+          break if ret != :redraw
```

Comment:
Maybe we could return different redraws or reload based on filesystem or partition changes. Them are the most important ones that affects all the widgets at all. If one partition is selected that does not support to be formatted or not support the current selected filesystem it implies modifications in all. 

In case of partition, I do a litle trick in the `FormatOptions` widget, storing the value and setting the options for that partition before when modified.